arm_single_step.patch

From: Christian Prochaska <christian.prochaska@genode-labs.com>

Enable software single-stepping on ARM.
---
 gdb/arm-tdep.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdb/arm-tdep.c b/gdb/arm-tdep.c
index be342117..8c87f098 100644
--- a/gdb/arm-tdep.c
+++ b/gdb/arm-tdep.c
@@ -9339,6 +9339,8 @@ arm_gdbarch_init (struct gdbarch_info info, struct gdbarch_list *arches)
   tdep = XCNEW (struct gdbarch_tdep);
   gdbarch = gdbarch_alloc (&info, tdep);
 
+  set_gdbarch_software_single_step(gdbarch, arm_software_single_step);
+
   set_solib_svr4_fetch_link_map_offsets(gdbarch, svr4_ilp32_fetch_link_map_offsets);
 
   /* Record additional information about the architecture we are defining.
